Naughts_and_Crosses.py

```diff
@@ -0,0 +1,227 @@
+
+    def printboard(self): #Simply prints an empty list into a matrix
+        for item in self.board_state:
+                print('|'.join(item))
+
+    def check_free_spaces(self): #Checks if there are any empty spaces left on the board
+        if self.empty_spaces == []:
+            return False
+        return True
+
+class Game():
+    running = True
+
+    def __init__(self): #Generates the board, then the players, then runs the game
+        self.board = Board()
+        self._generate_players()
+        self.run_game()
+
+    def run_game(self): #Runs the game by printing the board, asking who goes first, then loops the turn function until
+        self.board.printboard()  #the end of the game
+        self.first_turn()
+        while self.running == True:
+            self.turn()
+            self.board.printboard()
+            if self.check_win()
+            if self.check_draw()
+            self.change_player()
+
+    def first_turn(self): #Finds out who the user would like to go first
+        who_goes_first = input("Who should go first? (1/2)  ")
+        if who_goes_first == '1':
+            self.current_player = self.p1
+        elif who_goes_first == '2':
+            self.current_player = self.p2
+        else:
+            print("That's not a player number...")
+            self.first_turn()
+
+    def turn(self): #Checks the current player, makes them do a turn based on what type of player they are then
+        if self.current_player.player_type == 'H':  #takes the slot they used out of the empty spaces list
+            choices = self.current_player.human_choice()
+        else:
+            choices = self.current_player.ai_choice()
+        if choices in self.board.empty_spaces:
+            self.board.board_state[choices[0]][choices[1]] = self.current_player.symbol
+            self.board.empty_spaces.remove(choices)
+        else:
+            self.current_player.player_type == 'H':
+                print("That's already taken douche...")
+            self.turn()
+
+    def change_player(self): #Checks which players turn it is and then switches it then tells the next person who's go
+        if self.current_player == self.p1: # it is
+            self.current_player = self.p2
+        else:
+            self.current_player = self.p1
+        print('It is now ' + self.current_player.name + "'s go")
+
+    def check_draw(self): #Checks the empty spaces list and if there is no empty spaces, it's a draw
+        if not self.board.check_free_spaces():
+            self.running = False
+            print("Well great. That's a draw.")
+            self.end_game()
+
+    def check_win(self): #Checks the board to see if there is a player symbol in all the spaces of each win condition
+        for i in range((self.board.board_size * 2) + 2):
+        	winning_line = self.board.win_condition[i][j]
+                if all(
+                    self.board.board_state[winning_line[0]][winning_line[1]] 
```

Comment:
Same comment as last time, if you didn't attempt this as a one line for loop then it'll be neater:target: